includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -1048,6 +1058,53 @@ public function getAsJson() {
 		$this->setResponseData( $result, '', self::SIMPLE_JSON_VARNISH_CACHE_EXPIRATION );
 	}
 
+	/**
+	 * Get all top level sections as an array by splitting on h2 tags
+	 * @param $content
+	 * @return array
+	 */
+	private function splitArticleIntoSections( $content ) {
+		$pattern = '/(\<h2[^\>]*>)/';
```

Comment:
I feel like this leaves room for too many edge cases? Anyway we can make this stricter? How about h2s with id attributes?